.github/workflows/upf-unit-tests-pfcp.yml

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2025 Canonical Ltd.
+name: Unit tests PFCP
+
+on:
+  workflow_call:
+    inputs:
+      branch_name:
+        description: "Name of the branch to checkout"
+        required: false
+        type: string
+        default: ${{ github.ref }}
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch_name }}
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Unit tests
+        run: go test ./pfcpiface ./cmd/...
```

Comment:
What about reusing (and enhancing) the `unit-test.yml` action instead of creating another pretty similar action. What might be needed is to add another input parameter (e.g., directory) as an optional input parameter.